aws-iot-thing/src/main/java/software/amazon/iot/thing/Translator.java

```diff
@@ -40,61 +45,27 @@
  * mapping exceptions to appropriate Cloudformation exceptions
  */
 public class Translator {
-    static ProgressEvent<ResourceModel, CallbackContext> translateExceptionToProgressEvent(
-            ResourceModel model, Exception e, Logger logger) {
 
-        HandlerErrorCode errorCode = translateExceptionToErrorCode(e, logger);
-        ProgressEvent<ResourceModel, CallbackContext> progressEvent =
-                ProgressEvent.<ResourceModel, CallbackContext>builder()
-                        .resourceModel(model)
-                        .status(OperationStatus.FAILED)
-                        .errorCode(errorCode)
-                        .build();
-        if (errorCode != HandlerErrorCode.InternalFailure) {
-            progressEvent.setMessage(e.getMessage());
-        }
-        return progressEvent;
-    }
-
-
-    static HandlerErrorCode translateExceptionToErrorCode(Exception e, Logger logger) {
-        logger.log(String.format("Translating exception \"%s\", stack trace: %s",
-                e.getMessage(), ExceptionUtils.getStackTrace(e)));
-
-        // We're handling all the exceptions documented in API docs
-        // https://docs.aws.amazon.com/iot/latest/apireference/API_CreateThing.html#API_CreateThing_Errors
-        // (+same pages for other APIs)
-        // For Throttling and InternalFailure, we want CloudFormation to retry, and it will do so based on the error code.
-        // Reference with Retryable/Terminal in comments for each: https://tinyurl.com/y378qdno
+    static BaseHandlerException translateIotExceptionToHandlerException(
+            String resourceIdentifier, String operationName, IotException e) {
         if (e instanceof ResourceAlreadyExistsException) {
-            return HandlerErrorCode.AlreadyExists;
-        } else if (e instanceof software.amazon.cloudformation.exceptions.ResourceAlreadyExistsException) {
-            return HandlerErrorCode.AlreadyExists;
-        } else if (e instanceof InvalidRequestException) {
-            return HandlerErrorCode.InvalidRequest;
-        } else if (e instanceof LimitExceededException) {
-            return HandlerErrorCode.ServiceLimitExceeded;
-        } else if (e instanceof UnauthorizedException) {
-            return HandlerErrorCode.AccessDenied;
+            return new CfnAlreadyExistsException(ResourceModel.TYPE_NAME, resourceIdentifier, e);
+        } else if (e instanceof UnauthorizedException || e instanceof ResourceNotFoundException) {
```

Comment:
is `UnauthorizedException` supposed to be here? Doesnt seem right if we are returning `CfnNotFoundException `